packages/editor-ui/src/components/CredentialEdit/CredentialSharing.ee.vue

```diff
@@ -162,15 +162,19 @@ export default defineComponent({
 			return this.credentialsStore.getCredentialOwnerNameById(`${this.credentialId}`);
 		},
 		isCredentialSharedWithCurrentUser(): boolean {
-			return (this.credentialData.sharedWithProjects ?? []).some((sharee: IUser) => {
+			return (
+				Array.isArray(this.credentialData.sharedWithProjects)
+					? this.credentialData.sharedWithProjects
+					: []
+			).some((sharee) => {
 				return sharee.id === this.usersStore.currentUser?.id;
 			});
 		},
 		projects(): ProjectListItem[] {
 			return this.projectsStore.personalProjects.filter(
 				(project) =>
 					project.id !== this.credential?.homeProject?.id &&
-					project.id !== this.credentialData?.homeProject?.id,
+					project.id !== (this.credentialData?.homeProject as IDataObject)?.id,
```

Comment:
could we define this better in the original object? I know cred types are messy, but we should avoid bandaid solutions here.

Reply:
Updated.